### ISSUE TITLE
Fails to turn into correct MathJSON

### ISSUE THREAD

**Issue by stefnotch** [cortex-js/compute-engine#13]
I went through some mathematics notes and found the following interesting equations. I tested them on the compute engine demo page https://cortexjs.io/compute-engine/demo/

(Information for myself: Page 1 of ANA_UE_SS_21_Mo_1.pdf , meaning there are a *few* more testcases)

- [ ] `\begin{equation*} N(\varepsilon)\coloneq\lceil\frac{4}{\varepsilon^2}\rceil \end{equation*}`
![image](https://user-images.githubusercontent.com/10220080/124075310-f3eda080-da44-11eb-8bb9-01830e513dc3.png)
turns into `"" syntax-error`

- [ ] `\begin{equation*} x_{1,2}=1,2 \end{equation*}`
![image](https://user-images.githubusercontent.com/10220080/124075473-2f886a80-da45-11eb-92dc-f3417c9f3fc3.png)
turns into `["Equal", ["Subscript", "x",     {num: "12"}],   {num: "12"}]`. Notice how the comma is missing

- [ ] `\begin{equation*} \{1,2\} \end{equation*}`
![image](https://user-images.githubusercontent.com/10220080/124075660-7d9d6e00-da45-11eb-81c6-02d5ddf488f5.png)
turns into `["Multiply", "\\{",   {num: "12"}, "\\}"]`. I assume this happens because sets haven't been implemented yet

- [ ] `\begin{equation*} [1,2] \end{equation*}`
![image](https://user-images.githubusercontent.com/10220080/124075742-9c036980-da45-11eb-8566-b2a461a39125.png)
turns into `"" syntax-error`. I assume this happens because vectors/matrices and the like haven't been implemented yet.

<br>
<br>

- [ ] `\begin{equation*} \frac{2}{\sqrt{n}}\Leftrightarrow n>\frac{5}{n^2} \end{equation*}`
![image](https://user-images.githubusercontent.com/10220080/124076132-23e97380-da46-11eb-919d-559897371eec.png)
turns into the following, notice how `\\Leftrightarrow` doesn't have a special meaning and its precedence is off
```
[
  "Greater",
  [
    "Multiply",
    ["Divide", 
      {num: "2"}, ["Sqrt", "n"]],
    "\\Leftrightarrow",
    "n"
  ],
  ["Divide", 
    {num: "5"}, ["Power", "n", 2]]
]
```

<br>
<br>

- [ ] `\begin{equation*} |a_n|\le\frac{2}{\sqrt{n}}\Rightarrow a_n\to0=0 \end{equation*}`
![image](https://user-images.githubusercontent.com/10220080/124076357-6dd25980-da46-11eb-951f-3fcc85fb30be.png)
turns into the following, notice how `\\Rightarrow` doesn't have a special meaning (implies) and its precedence is off. 
```
[
  "LessEqual",
  ["Abs", ["Subscript", "a", "n"]],
  [
    "Equal",
    [
      "To",
      [
        "Multiply",
        ["Divide", 
          {num: "2"}, ["Sqrt", "n"]],
        "\\Rightarrow",
        ["Subscript", "a", "n"]
      ],
      
      {num: "0"}
    ],
    
    {num: "0"}
  ]
]
```

**Comment by arnog** [comment]
Those are excellent test cases, thank you!

`x_{1,2}` is a particularly interesting one: the comma is a valid inter-digit separator (as in "1,234.56") but of course I should be a bit more rigorous when considering it as a digit separator. Still, I suppose that `x_123,456` would be ambiguous... Hmmm..

**Comment by stefnotch** [comment]
That's true, that makes this test case surprisingly tricky.
It doesn't help that countries and standards can't agree on which decimal separator to use https://en.wikipedia.org/wiki/Decimal_separator
(My preferred one is grouping digits into groups of 3. Simple spacing rules and no funky, ambiguous extra characters.)

For what it's worth, I personally have never seen a `,` decimal separator in a mathematical equation. However, it does appear somewhat frequently in tables of numbers and other cases where it's mostly just a number. 

Another interesting thing that I just found out is that by default, a comma has a small space after it. 
![image](https://user-images.githubusercontent.com/10220080/124082665-3fa44800-da4d-11eb-976a-34f06905e048.png)

**Comment by arnog** [comment]
Yes, it's considered a punctuation character and so has a small padding on its right.

Note also that you can configure which character (or rather which LaTeX command sequence) is used with the `groupSeparator` option.

That said, some people have very strong opinions about this: http://wordpress.mrreid.org/2014/05/27/stop-putting-commas-in-your-numbers/

Maybe I should reconsider the default value for `groupSeparator`... :)

**Comment by stefnotch** [comment]
Oh nice, I didn't know that there was configuration option for that :)

I like the idea of changing the default value for `groupSeparator`, but it doesn't matter very much to me either way.

**Comment by stefnotch** [comment]
I found another case where the compute-engine doesn't quite work: Logarithms.

For example `\begin{equation*} \log_3(7) \end{equation*}` gets parsed as
```
[
  "Multiply",
  ["Subscript", "\\log", 3],
  ["Parentheses", 
    {num: "7"}]
]
```
